src/Stoa.ts

```diff
@@ -173,8 +173,16 @@ class Stoa extends WebService
                     }
                     else
                     {
-                        preimage_distance = NaN;
-                        result_preimage_hash = Hash.NULL;
+                        if (target_height == new Height(row.enrolled_at))
+                        {
+                            preimage_distance = 0;
```

Comment:
How does the re-enrollment change the preimage distance?
I'll have to find out.

Reply:
Re-enrolling sets the distance to zero.